### PR TITLE
Added vanilla ratios to the freezer

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description
Added transformation for ice that follows vanilla rules

## Proposed changes

a simple if statement that checks for the item setting

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [x] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 1]
```diff
@@ -17,16 +18,25 @@
...
     }
 
     @Override
     protected void registerDefaultRecipes() {
+        // This if statement makes the transformation follow Minecraft logic
+        if (vanillaTransformation.getValue()){
+            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE, 9) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
+            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE, 9) }, new ItemStack[] { new ItemStack(Material.BLUE_ICE) });
+        } else {
+            registerRecipe(4, new ItemStack[] {new ItemStack(Material.ICE)}, new ItemStack[] {new ItemStack(Material.PACKED_ICE)});
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:35]: Formatting
```suggestion
            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 2]
```diff
@@ -17,16 +18,25 @@
...
     @Override
     protected void registerDefaultRecipes() {
+        // This if statement makes the transformation follow Minecraft logic
+        if (vanillaTransformation.getValue()) {
+            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE, 9) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
+            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE, 9) }, new ItemStack[] { new ItemStack(Material.BLUE_ICE) });
+        } else {
+            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE) }, new ItemStack[] {new ItemStack(Material.PACKED_ICE)});
+            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE) }, new ItemStack[] {new ItemStack(Material.BLUE_ICE)});
+        }
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:37]: here
```suggestion
            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE) }, new ItemStack[] { new ItemStack(Material.BLUE_ICE) });
        }
```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java (hunk 2 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java]: why are you reviewing old stuff?

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 3]
```diff
@@ -17,16 +18,25 @@
...
     @Override
     protected void registerDefaultRecipes() {
+        // This if statement makes the transformation follow Minecraft logic
+        if (vanillaTransformation.getValue()) {
+            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE, 9) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
+            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE, 9) }, new ItemStack[] { new ItemStack(Material.BLUE_ICE) });
+        } else {
+            registerRecipe(4, new ItemStack[] { new ItemStack(Material.ICE) }, new ItemStack[] { new ItemStack(Material.PACKED_ICE) });
+            registerRecipe(6, new ItemStack[] { new ItemStack(Material.PACKED_ICE) }, new ItemStack[] { new ItemStack(Material.BLUE_ICE) });
+        }
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:37]: new line
```suggestion
        }

```

**Review by WalshyDev — CHANGES_REQUESTED** [review]

**Review by WalshyDev — DISMISSED** [review]
LGTM I guess

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 4]
```diff
@@ -17,16 +18,26 @@
 
 public class Freezer extends AContainer implements RecipeDisplayItem {
 
+    private final ItemSetting<Boolean> vanillaTransformation = new ItemSetting<>(this, "vanilla-translation", false);
+
     public Freezer(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:23]: Can you add a Params are nonnull by default annotation here?

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 5]
```diff
@@ -17,16 +18,26 @@
 
 public class Freezer extends AContainer implements RecipeDisplayItem {
 
+    private final ItemSetting<Boolean> vanillaTransformation = new ItemSetting<>(this, "vanilla-translation", false);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:21]: I think this setting needs a better name, e.g. `use-vanilla-ice-ratio`.
Although that might make people think of vanilla ice cream XD

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 6]
```diff
@@ -17,16 +18,26 @@
 
 public class Freezer extends AContainer implements RecipeDisplayItem {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:19]: Can this class have javadocs please?
Not in the scope of this PR, I know, just pointing it out

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java (hunk 5 above)
Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:21]: it depends if more is added in the future... `use-vanilla-ratio` would be better

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:21]: `use-vanilla-ratios` then

Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:21]: Do you want me to leave a comment saying dont get confused by ice cream? XD

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 7]
```diff
@@ -15,18 +17,38 @@
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineRecipe;
 
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * The {@link Freezer} can freeze items into it's frozen state.
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:23]: "its" not "it's"
```suggestion
 * The {@link Freezer} can freeze items into its frozen state, e.g. water to ice.
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]
One small mistake, otherwise good

**Review by TheBusyBiscuit — DISMISSED** [review]

**Comment by svr333** [comment]
thanks for putting me as author though Idk what I did

**Review by svr333 — APPROVED** [review]

**Comment by J3fftw1** [comment]
In the gitblame you showed up for changing something. I didnt want to have just my name there so took everyone who showed up in git blame

**Comment by svr333** [comment]
ah fair enough

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java [hunk 8]
```diff
@@ -4,9 +4,11 @@
 import java.util.List;
 
 import org.bukkit.Material;
+
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:7]: Random spacing...

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java (hunk 8 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/Freezer.java:7]:
```suggestion
```

**Review by WalshyDev — APPROVED** [review]
LGTM

**Review by svr333 — APPROVED** [review]

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**